### PR TITLE
Splitimage

### PR DESCRIPTION
**Pull request by montjoie**

### CODE REVIEW

File: lava-docker-master-image/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
+FROM bitnami/minideb:stretch
+
+RUN echo "deb http://ftp.fr.debian.org/debian/ stretch-backports main" >> /etc/apt/sources.list
```
Comment (khilman) [inline lava-docker-master-image/Dockerfile]: Please drop hard-coded .fr mirrors.

File: lava-slave/Dockerfile [hunk 2]
```diff
@@ -1,40 +1,4 @@
...
-#RUN DEBIAN_FRONTEND=noninteractive apt-get -y install wget gnupg \
-# && wget http://images.validation.linaro.org/production-repo/production-repo.key.asc \
-# && apt-key add production-repo.key.asc \
-# && echo 'deb http://images.validation.linaro.org/production-repo/ stretch-backports main' > /etc/apt/sources.list.d/lava.list \
-# && apt-get clean && apt-get update
-#RUN DEBIAN_FRONTEND=noninteractive apt-get -y install lava-dispatcher
-
-COPY scripts/build-lava /
-RUN sh /build-lava lava-dispatcher 2018.4 2018.4-1
+FROM montjoie/lava-docker-slave-image:latest
```
Comment (khilman) [inline lava-slave/Dockerfile]: Let's use the baylibre docker hub for this for the final version.

File: lava-docker-slave-image/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+RUN \
+ echo 'lava-server   lava-server/instance-name string lava-slave-instance' | debconf-set-selections && \
+ echo 'locales locales/locales_to_be_generated multiselect C.UTF-8 UTF-8, en_US.UTF-8 UTF-8 ' | debconf-set-selections && \
+ echo 'locales locales/default_environment_locale select en_US.UTF-8' | debconf-set-selections && \
+ DEBIAN_FRONTEND=noninteractive apt-get -y install \
+ locales \
+ vim \
+ sudo \
+ cu \
+ conmux \
```
Comment (khilman) [inline lava-docker-slave-image/Dockerfile]: cu and conmux aren't strictly needed for the slave image, right?  Those should probably be in the overlay images.

File: lava-docker-master-image/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ && apt-key add production-repo.key.asc \
+ && echo 'deb http://images.validation.linaro.org/production-repo/ stretch-backports main' > /etc/apt/sources.list.d/lava.list \
+ && apt-get clean && apt-get update
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install \
+ qemu-system \
+ qemu-system-arm \
+ qemu-system-i386 \
+ ser2net \
+ u-boot-tools \
```
Comment (khilman) [inline lava-docker-master-image/Dockerfile]: I don't think ser2net or u-boot-tools are needed on the master.

Comment (khilman) [reply lava-docker-master-image/Dockerfile]: For that matter, I don't think the QEMU stuff is needed either when the master is a "dumb" master (no devices.)

**Review by khilman — CHANGES_REQUESTED** [review]

**Comment by khilman** [comment]
Some bike-shedding for the naming.
I'd prefer:
s/lava-docker-master-image/lava-master-base-image/
s/lava-docker-slave-image/lava-slave-base-image/

File: lava-docker-master-image/Dockerfile (hunk 1 above)
Comment (montjoie) [reply lava-docker-master-image/Dockerfile]: done

File: lava-slave/Dockerfile (hunk 2 above)
Comment (montjoie) [reply lava-slave/Dockerfile]: done

File: lava-docker-slave-image/Dockerfile (hunk 3 above)
Comment (montjoie) [reply lava-docker-slave-image/Dockerfile]: cu and conmux moved

File: lava-docker-master-image/Dockerfile (hunk 4 above)
Comment (montjoie) [reply lava-docker-master-image/Dockerfile]: removed from master

File: lava-master/Dockerfile [hunk 5]
```diff
@@ -1,79 +1,13 @@
...
-# lava \
-# && /stop.sh
-
-# Install latest
-COPY scripts/build-lava /
-RUN bash /build-lava lava-dispatcher 2018.4 2018.4-1
-RUN /etc/init.d/postgresql start && \
- bash /build-lava lava-server 2018.4 2018.4-1 && \
- /etc/init.d/postgresql stop
+FROM baylibre/lava-master-docker-image:latest
```
Comment (khilman) [inline lava-master/Dockerfile]: Please rename to baylibre/lava-master-base, and also add tags to Docker hub for lava versions.  Then use specific tag here.

File: lava-slave/Dockerfile [hunk 6]
```diff
@@ -1,40 +1,6 @@
-FROM bitnami/minideb:stretch
+FROM baylibre/lava-slave-docker-image:latest
```
Comment (khilman) [inline lava-slave/Dockerfile]: Similar to master, please rename to baylibre/lava-slave-base and use specific version.

File: lava-master-docker-image/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+COPY scripts/lava-master /etc/init.d/
+RUN chmod 755 /etc/init.d/lava-master
+COPY scripts/lava-slave /etc/init.d/
+RUN chmod 755 /etc/init.d/lava-slave
+COPY scripts/lava-server-gunicorn /etc/init.d/
+RUN chmod 755 /etc/init.d/lava-server-gunicorn
+
+RUN dpkg -l | grep lava
+
+EXPOSE 69/udp 80 3079 5555 5556
```
Comment (khilman) [inline lava-master-docker-image/Dockerfile]: To fix in your fixes PR: the master doesn't need to expose TFTP port.

File: lava-master/Dockerfile (hunk 5 above)
Comment (khilman) [reply lava-master/Dockerfile]: Actually, we don't need the base either.  Just baylibre/lava-master is enough.

File: lava-slave/Dockerfile (hunk 6 above)
Comment (khilman) [reply lava-slave/Dockerfile]: use: baylibre/lava-slave

**Review by khilman — CHANGES_REQUESTED** [review]

File: lava-master/Dockerfile (hunk 5 above)
Comment (montjoie) [reply lava-master/Dockerfile]: as discussed in meeting, changed to lava-master-base:latest

File: lava-slave/Dockerfile (hunk 6 above)
Comment (montjoie) [reply lava-slave/Dockerfile]: as discussed in meeting, changed to lava-slave-base:latest

File: lava-master-docker-image/Dockerfile (hunk 7 above)
Comment (montjoie) [reply lava-master-docker-image/Dockerfile]: done

**Comment by khilman** [comment]
Something is wrong as this shows 40 patches on top of master, when I think there should only be 2?

**Pull request merged**